lib/balance-tx/lib/internal/Internal/Cardano/Write/Tx.hs

```diff
@@ -833,27 +839,44 @@ evaluateTransactionBalance
     -> Core.TxBody era
     -> Core.Value era
 evaluateTransactionBalance pp utxo =
-    let -- Looks up the current deposit amount for a registered stake credential
-        -- delegation.
+    let
+        -- Deposit lookup for 'DeRegKey' delegation certificates in the TxBody
         --
-        -- This function must produce a valid answer for all stake credentials
-        -- present in any of the 'DeRegKey' delegation certificates in the
-        -- supplied 'TxBody'. In other words, there is no requirement to know
-        -- about all of the delegation certificates in the ledger state,
-        -- just those this transaction cares about.
-        lookupRefund :: Core.StakeCredential StandardCrypto -> Maybe Coin
-        lookupRefund _stakeCred = Just $ pp ^. Core.ppKeyDepositL
+        -- TODO [ADP-3270] Query actual value of deposit
+        --
+        -- https://cardanofoundation.atlassian.net/browse/ADP-3270
+        keyDepositAssumeCurrent
+            :: Core.StakeCredential StandardCrypto -> Maybe Coin
+        keyDepositAssumeCurrent _stakeCred = Just $ pp ^. ppKeyDepositL
+
+        -- Deposit lookup for `UnRegDRep` certificates in the TxBody
+        --
+        -- TODO [ADP-3270] Query actual value of deposit
+        --
+        -- https://cardanofoundation.atlassian.net/browse/ADP-3270
+        dRepDepositAssumeCurrent
+            :: Core.Credential 'Ledger.DRepRole StandardCrypto
+            -> Maybe Coin
+        dRepDepositAssumeCurrent _drepCred = case recentEra @era of
+            RecentEraConway -> Just $ pp ^. ppDRepDepositL
+            RecentEraBabbage -> error "impossible: lookupDRep called in Babbage"
 
         -- Checks whether a pool with a supplied 'PoolStakeId' is already
         -- registered.
         --
         -- There is no requirement to answer this question for all stake pool
         -- credentials, just those that have their registration certificates
         -- included in the supplied 'TxBody'.
-        isRegPoolId :: Ledger.KeyHash 'Ledger.StakePool StandardCrypto -> Bool
-        isRegPoolId _keyHash = True
-
-    in Ledger.evalBalanceTxBody pp lookupRefund isRegPoolId utxo
+        assumePoolIsReg :: Ledger.KeyHash 'Ledger.StakePool StandardCrypto -> Bool
```

Comment:
Overly-long line. ^^^

We could do:
```suggestion
        assumePoolIsReg
            :: Ledger.KeyHash 'Ledger.StakePool StandardCrypto
            -> Bool
```

But we could also save 2 lines and an entire indentation level (4 spaces) by moving the definitions to a `where` clause, which also removes the need to wrap:
```hs
evaluateTransactionBalance pp utxo =
    Ledger.evalBalanceTxBody
        pp
        keyDepositAssumeCurrent
        dRepDepositAssumeCurrent
        assumePoolIsReg
        utxo
  where
    -- Deposit lookup for 'DeRegKey' delegation certificates in the TxBody
    --
    -- TODO [ADP-3270] Query actual value of deposit
    --
    -- https://cardanofoundation.atlassian.net/browse/ADP-3270
    keyDepositAssumeCurrent
        :: Core.StakeCredential StandardCrypto -> Maybe Coin
    keyDepositAssumeCurrent _stakeCred = Just $ pp ^. ppKeyDepositL

    -- Deposit lookup for `UnRegDRep` certificates in the TxBody
    --
    -- TODO [ADP-3270] Query actual value of deposit
    --
    -- https://cardanofoundation.atlassian.net/browse/ADP-3270
    dRepDepositAssumeCurrent
        :: Core.Credential 'Ledger.DRepRole StandardCrypto
        -> Maybe Coin
    dRepDepositAssumeCurrent _drepCred = case recentEra @era of
        RecentEraConway -> Just $ pp ^. ppDRepDepositL
        RecentEraBabbage -> error "impossible: lookupDRep called in Babbage"

    -- Checks whether a pool with a supplied 'PoolStakeId' is already
    -- registered.
    --
    -- There is no requirement to answer this question for all stake pool
    -- credentials, just those that have their registration certificates
    -- included in the supplied 'TxBody'.
    assumePoolIsReg :: Ledger.KeyHash 'Ledger.StakePool StandardCrypto -> Bool
    assumePoolIsReg _keyHash = True
```